mender-auth/ipc/client_test.cpp

```diff
@@ -0,0 +1,226 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <mender-auth/ipc/client.hpp>
+
+#include <thread>
+#include <string>
+#include <iostream>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <common/conf.hpp>
+#include <common/http.hpp>
+#include <common/io.hpp>
+#include <common/path.hpp>
+#include <common/testing.hpp>
+#include <vector>
+
+#define TEST_PORT "8001"
+#define TEST_PORT_2 "8002"
+
+using namespace std;
+
+namespace conf = mender::common::conf;
+namespace error = mender::common::error;
+namespace http = mender::http;
+namespace io = mender::common::io;
+namespace path = mender::common::path;
+namespace mtesting = mender::common::testing;
+
+namespace ipc = mender::ipc;
+
+using TestEventLoop = mender::common::testing::TestEventLoop;
+
+class ListenClientTests : public testing::Test {
```

Comment:
The tests here are a little different.

1. is run in threads
2. The other has all the functionality in one loop, which I expect to be racy?

How should we ideally handle tests like this with multiple servers, setup, timing and ordering requirements?

Reply:
You are using separate objects in each thread, so I can't immediately see why it should be racy. To me it looks ok (we are doing something similar in the Timers test to cut down on test time, btw).

If you need ordering, something like what I did [here](https://github.com/mendersoftware/mender/commit/1bbb3856e5c0ef433dc0953e93fdb26d5457f1a2) is a good way.